source/module_base/test/complexarray_test.cpp

```diff
@@ -167,3 +290,148 @@ TEST_F(ComplexArray_test,operator_parentheses)
     EXPECT_DEATH(a2(1,1,1,0),"");
 }
 
+TEST_F(ComplexArray_test,fuction_zero_out)
+{
+    a2.zero_out();
+    for (int i=0;i<a2.getSize();++i)
+    {
+        EXPECT_EQ(a2.ptr[i],comzero);
+    }
+}
+
+TEST_F(ComplexArray_test,fuction_negate)
+{
+    a2.negate();
+    for (int i=0;i<a2.getSize();++i)
+    {
+        EXPECT_EQ(a2.ptr[i],com1*-1);
+    }
+}
+
+TEST(ComplexArray,fuction_randomize)
+{
+    ModuleBase::ComplexArray a(10,10,10,10);
+    a.randomize();
+
+    bool allequal = true;
+    for (int i=1;i<5;++i)
+    {
+        allequal = allequal && (a.ptr[i] == a.ptr[0]);
+    }
+    EXPECT_FALSE(allequal);
+
+    for (int i=0;i<a.getSize();++i)
+    {
+        EXPECT_TRUE((a.ptr[i].real() >= -0.5) && (a.ptr[i].imag() >= -0.5));
+        EXPECT_TRUE((a.ptr[i].real() < 0.5) && (a.ptr[i].imag() < 0.5));
```

Comment:
What's the difference between line 325 and 326?

Reply:
Isn't that obvious?

Reply:
> Isn't that obvious?

My bad, I didn't see the minus.